docs/vsphere_ci/README.md

```diff
@@ -0,0 +1,178 @@
+         ],
+         "type":"vsphere-iso",
+         "username":"{{user `vsphere-user`}}",
+         "vcenter_server":"{{user `vsphere-server`}}",
+         "vm_name":"{{user `vm-name`}}",
+         "ssh_password":"{{user `winadmin-password`}}",
+         "ssh_username":"Administrator",
+         "ssh_timeout":"50m"
+      }
+   ],
+   "provisioners":[
+      {
+	"pause_before": "120s",
+	"inline":["ipconfig"],
+        "type":"powershell",
+	"elevated_user": "Administrator",
+        "elevated_password": "{{user `winadmin-password`}}",
+	"max_retries":"10"
+      }
+   ],
+   "sensitive-variables":[
+      "vsphere_password",
+      "winadmin_password"
+   ],
+   "variables":{
+      "os_iso_path":"<Windows_os_iso_path>",
+      "vm-cpu-num":"2",
+      "vm-disk-size":"128000",
+      "vm-mem-size":"4096",
+      "vm-name":"<template_to_be_created>",
+      "vsphere-cluster":"Cluster-1",
+      "vsphere-datacenter":"<datacenter_name>",
+      "vsphere-datastore":"<datastore_name>",
+      "vsphere-folder":"<folder_where_template_gets_built>",
+      "vsphere-network":"network_segment_name",
+      "vsphere-password":"<password>",
+      "vsphere-server":"<server_name>",
+      "vsphere-user":"<user_name>",
+      "winadmin-password":"<Windows_os_password>"
+   }
+}
+```
+
+### Variables
+- `<tools_iso_path>` - Path where VMWare tools iso is available in vSphere datacenter
+- `<Windows_os_iso_path>` - Path where Windows ISO was dowloaded to in vSphere datacenter
+- `<template_to_be_created>` - Name of the Windows vSphere template that will be created by successful Packer run in 
+ 			       vSphere datacenter
+- `<datacenter_name>` - Name of the vSphere datacenter
+- `<datastore_name>` - Name of the vSphere datastore
+- `<folder_where_template_gets_built>` - Path where Windows template gets built by successful Packer run in 
+					 vSphere datacenter
+- `<password>` - vSphere environment password
+- `<server_name>` - vSphere environment URL
+- `<user_name>` - vSphere username to login
+- `<Windows_os_password>` - Password for the Windows template
+
+## What actually happens during build
+
+Packer mounts the Windows iso and starts the VM. 
+- All the files in `floppy_files` section of your build file will be copied to the floppy disk of the mounted iso 
+ which is represented as `a:\` drive in the Windows VM
+- [Autounattend.xml](scripts/autounattend.xml) is a special file in Windows which gets automatically executed once the
+VM starts. You can specify all the commands that needs to executed on first boot.
+
+## Building with Packer
+
+Packer relies on a [build file](build.json) for VM image creation. 
+
+To build:
+
+`packer build build.json`
+
+To forcefully rebuild the template:
+
+`packer build -force build.json`
+
+To enable logging:
+
+`PACKER_LOG=1 packer build -force build.json`
+
+## Post Packer build
+
+Once the Packer build completes successfully, the template must be converted to a VM.
+
+![Convert to VM](images/VMConversion.png)
```

Comment:
There is a missing piece here. What gets plugged into the `MachineSet`?

Reply:
VM.